### DISCUSSION TITLE
How to display a table

### DISCUSSION THREAD

**Discussion by hilmanski** [spatie/laravel-markdown#41] [Q&A]
Let's say I want make a 'table' via markdown like this
```
| Free | Pro  |
| --- | --- |
| make your own |custom image |
```

Anyone know how to display it via this package?

**Comment by natecorkish** [comment]
In your markdown.php 

Add:

```
'extensions' => [
    new \League\CommonMark\Extension\Table\TableExtension(),
],
```

**Reply by 3rgo** [reply]
This is not working for me.
 I've tried clearing the application cache, and even disabled caching in the extension config to ensure it was not interfering. Everything else in my markdown file (bold, italic, titles) renders properly.

EDIT : this is not working when using the blade `<x-markdown>` component inside a livewire component (markdown content stored in a public livewire property). If the property contains HTML obtained through the [container renderer](https://spatie.be/docs/laravel-markdown/v1/rendering-markdown/general-usage), it works though.

**Reply by jimmitjoo** [reply]
@3rgo This is actually still the case. Tried work around it, but no success so going with the container renderer as well.

**Reply by gemmadlou** [reply]
Just running into this problem now. Taking a look.

**Reply by gemmadlou** [reply]
Yeah - same issue using the x-markdown component. This worked albeit, manually.

```php
echo app(Spatie\LaravelMarkdown\MarkdownRenderer::class)->toHtml("
|Country|Medical Certificate Required|Notes|
|:-|:-|:-|
|United Kingdom|Yes|[www.exteriores.gob.es](http://www.exteriores.gob.es), Over 180 days. Certificate must be from the UK or Spain.|
");
?>
```

I might make a PR for it if I get the time.

**Reply by reaves123** [reply]
Anyone succeeded with tables yet?
I am also stuck at this point right now.

**Reply by rabol** [reply]
any solution ?

**Comment by codeldev** [comment]
I had expected it to render tables by default, but it seems the TableExtension is required. This works fine:

markdown.php config

```php
'extensions' => [
   League\CommonMark\Extension\Table\TableExtension::class,
],
```

You will also need to add the table options:

```php
'commonmark_options' => [
    'table' => [
        'wrap' => [
            'enabled' => false,
            'tag' => 'div',
            'attributes' => [],
        ],
        'alignment_attributes' => [
            'left'   => ['align' => 'left'],
            'center' => ['align' => 'center'],
            'right'  => ['align' => 'right'],
        ],
    ],
],
```

Hope that helps

**Reply by robinmoisson** [reply]
Awesome, thanks!
